config/config.go

```diff
@@ -93,9 +93,13 @@ func NewDefaultAgentConfig() *AgentConfig {
 		// This is a hardcoded URL so parsing it should not fail
 		panic(err)
 	}
+
+	_, err = container.GetContainers()
+	canAccessContainers := err == nil
+
 	ac := &AgentConfig{
 		// We'll always run inside of a container.
-		Enabled:       container.IsAvailable(),
+		Enabled:       canAccessContainers,
```

Comment:
`container.IsAvailable()` actually just looks to see if a `listener` in the configuration was set. It makes more sense to try and connect to the container provider to determine if this should be enabled.